src/hist.jl

```diff
@@ -115,6 +115,37 @@ end
 abstract type AbstractHistogram{T<:Real,N,E} end
 
 # N-dimensional histogram object
+"""
+    Histogram <: AbstractHistogram
+
+The `Histogram` type represents data that has been tabulated into intervals
+(known as *bins*) along the real line, or in higher dimensions, over a real space.
+Histograms can be fitted to data using the `fit` method.
+
+# Fields
+* edges: An iterator that contains the boundaries of the bins in each dimension.
+* weights: An array that contains the weight of each bin.
+* closed: A symbol with value `:right` or `:left` indicating on which side bins
+  (which half-open intervals or higher-dimensional analogues thereof) are closed.
+```jldoctest
+julia> fit(Histogram, [2.],  1:3, closed=:left)
+Histogram{Int64,1,Tuple{UnitRange{Int64}}}
+edges:
+  1:3
+weights: [0, 1]
+closed: left
+isdensity: false
```

Comment:
That field should be documented too.

Reply:
I think I documented all fields, if you are missing `isdensity`, it is there, just a few lines below.

Reply:
Ah, then it should be above the example, not below.

Also, the description of `isdensity` doesn't sound precise enough to me: "amount" is used for both `true` and `false`. AFAIK when `true` the weights sum to unity? Or maybe it's more subtle than that?

Reply:
> Ah, then it should be above the example, not below.

Not sure. The example is about the `closed` field

> Also, the description of isdensity doesn't sound precise enough to me: "amount" is used for both true and false.

Mhh one is  "amount" and the other is "amount per volume". Can you reread the docstring and suggest an concrete improvement?

> AFAIK when `true` the weights sum to unity? Or maybe it's more subtle than that?

I though so too. But then my code had strange bugs and I discovered the real meaning of `isdensity`. In fact `isdensity` has little relation to whether the histogram is normalized.

Reply:
> Not sure. The example is about the `closed` field

Well, it breaks the list of arguments, and it also illustrates other fields (which don't have their own examples). AFAICT the convention is to put all examples at the end.

> Mhh one is "amount" and the other is "amount per volume". Can you reread the docstring and suggest an concrete improvement?

I'm not sure I can suggest an exact wording since as I said I don't know what this field means. But at least I can say that the description wasn't enough for me to understand that. :-) What is "volume" here? Also I'd say "amount of a quantity" is a pleonasm.

Maybe @simonbyrne can help.

Reply:
I will try an example. We have an area in which weasels live. The area is subdivided into a grid given by `edges`. Imagine non uniform grid, e.g `edges=([1,2,10], 1:3)`. We want a histogram of the weasel population and need to fill the `weights` matrix. One way is to fill the matrix with the numbers of weasels living in each cell. The other is to fill the matrix with the density of weasels in each cell. This first corresponds to `isdensity=true` the latter to `isdensity=false`.

Reply:
OK, thanks. Then how can we define "density" in general? I guess it is the weights divided by the width (1D) or area (>1D) of the bin?

Reply:
FWIW, [Numpy](https://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram.html) says this:
> density: If False, the result will contain the number of samples in each bin. If True, the result is the value of the probability density function at the bin, normalized such that the integral over the range is 1. Note that the sum of the histogram values will not be equal to 1 unless bins of unity width are chosen; it is not a probability mass function.

Reply:
By denisity I mean number of weasels living in the bin / area of the bin.

Reply:
Ah nice it did not occur to me to look at numpy. In our case `isdensity=true` does not mean that the integral is one (which is good I think).

Reply:
I added an example, you think it clarifies things @nalimilan ?